src/Widget/Controller/WidgetController.php

```diff
@@ -161,14 +161,14 @@ public function renderPage(Request $request, WidgetPageInterface $widgetPage, $f
      * @param $cdbid
      * @return JsonResponse
      */
-    public function renderWidget(Request $request, WidgetPageInterface $widgetPage, $widgetId, $cdbid = '')
+    public function renderWidget(Request $request, WidgetPageInterface $widgetPage, $widgetId, $cdbid = '', $preferredLanguage = 'nl')
```

Comment:
Is it possible you forgot to add this to the paths that point to this controller method?
I cannot find any usages of this parameter.

Reply:
@bertramakers I don't understand what you mean. Isn't it in my 1st commit?

Reply:
No, that's a different class?

This is the code from your first commit, in `WidgetApiController`:
```
$data['preview'] = $this->renderer->renderWidget($widget, '', $widgetPage->getLanguage());
```
In that one, `$this->renderer` is an instance of `RendererInterface`.

Now here in `WidgetController`, you've changed the signature of `WidgetController::renderWidget`.
That's not the `$this->renderer` used in `WidgetApiController` right?

If you look at [line 183](https://github.com/cultuurnet/projectaanvraag-silex/pull/169/files#diff-796c27d5b0a3ce49b2ecc5d257c6f92eL183), you can see that `WidgetController::renderWidget` method even uses the same `RendererInterface` underneath. It's just a different endpoint.

So what is calling this `WidgetController::renderWidget`?
According to PHPStorm, it's these three routes:
- https://github.com/cultuurnet/projectaanvraag-silex/blob/master/app/Widget/WidgetControllerProvider.php#L28
- https://github.com/cultuurnet/projectaanvraag-silex/blob/master/app/Widget/WidgetControllerProvider.php#L30
- https://github.com/cultuurnet/projectaanvraag-silex/blob/master/app/Widget/WidgetControllerProvider.php#L36

But none of these have a `{preferredLanguage}` parameter in their path names.

So this parameter will always be the default value, `nl`.

Reply:
You can also see the difference by comparing:
```
$this->renderer->renderWidget($widget, '', $widgetPage->getLanguage())
```
from the first commit, and 
```
public function renderWidget(Request $request, WidgetPageInterface $widgetPage, $widgetId, $cdbid = '', $preferredLanguage = 'nl')
```
from the second one.
The first method has no `$request` argument. So it's not calling the method from the second commit but the one on `RendererInterface`.

Reply:
To be clear the issue is not with the first commit, but with the extra parameter in the second commit. It's fine to add it, but it's not being set anywhere so it's always `nl`. So it's essentially not changing anything.

Reply:
@bertramakers thank you for the extensive explanation!
Fixed it now